### PR TITLE
Application state scout rover

### PR DESCRIPTION
**Pull request by nmarshak1337**
The scout rover is now in the dummy scenario. Plus the lighting bug has been fixed.

### CODE REVIEW

File: source/rover.js [hunk 1]
```diff
@@ -18,6 +18,18 @@ this.initialize = function() {
...
     }
 
+    //TODO: Set this properly (we shouldn't have to set this explicity in here...
+    // this.transform gets set automatically for ONE of the rovers, but not the other one)
+    this.transform = [
+        1,  0,  0,  0,
+        0,  1,  0,  0,
+        0,  0,  1,  0,
+        0,  0,  0,  1
+    ];
```
Comment (BrettASwift) [inline source/rover.js]: This has returned.

File: source/rover.js [hunk 2]
```diff
@@ -97,47 +109,67 @@ this.placeOnTerrain = function( pos ) {
...
+    // // Step 2: Find the normal of the plane on which the rover will sit
+    // var vecFLtoFR = goog.vec.Vec3.subtract( terrainPosFR, terrainPosFL, goog.vec.Vec3.create() );
+    // var vecFLtoBL = goog.vec.Vec3.subtract( terrainPosBL, terrainPosFL, goog.vec.Vec3.create() );
+    // var vecBRtoFR = goog.vec.Vec3.subtract( terrainPosFR, terrainPosBR, goog.vec.Vec3.create() );
+    // var vecBRtoBL = goog.vec.Vec3.subtract( terrainPosBL, terrainPosBR, goog.vec.Vec3.create() );
+    // var normal = this.calcUpwardNormalizedPlaneNormal( vecFLtoBL, vecFLtoFR );
+    // var normalEstimate2 = this.calcUpwardNormalizedPlaneNormal( vecBRtoFR, vecBRtoBL );
+    // normal[ 0 ] = ( normal[ 0 ] + normalEstimate2[ 0 ] ) * 0.5;
+    // normal[ 1 ] = ( normal[ 1 ] + normalEstimate2[ 1 ] ) * 0.5;
+    // normal[ 2 ] = ( normal[ 2 ] + normalEstimate2[ 2 ] ) * 0.5;
```
Comment (BrettASwift) [inline source/rover.js]: Lots of commented out code that can be removed.

File: source/rover.js [hunk 3]
```diff
@@ -97,47 +109,67 @@ this.placeOnTerrain = function( pos ) {
...
+    //     0, 0, 1, 0,
+    //     pos[ 0 ],    pos[ 1 ],    pos[ 2 ],    1
+    // ];
     this.transform = [
-        xAxis[ 0 ],  xAxis[ 1 ],  xAxis[ 2 ],  0,
-        yAxis[ 0 ],  yAxis[ 1 ],  yAxis[ 2 ],  0,
-        normal[ 0 ], normal[ 1 ], normal[ 2 ], 0,
+        this.transform[0],  this.transform[1],  this.transform[2],  0,
+        this.transform[4],  this.transform[5],  this.transform[6],  0,
+        this.transform[8], this.transform[9], this.transform[10], 0,
```
Comment (kadst43) [inline source/rover.js]: spacing inside the [].

**Comment by BrettASwift** [comment]
:+1:

**Pull request merged**